src/highlevel.jl

```diff
@@ -174,8 +174,31 @@ function (o::InferenceSession)(
             throw(ArgumentError(msg))
         end
     end
+    isalive(o) || error("Session has been released and can no longer be called.")
```

Comment:
Can we call this right at the beginning of this function? With the current version it is not dead obvious to me, whether we try to access deleted stuff before the `isalive` call.

Reply:
I have no strong opinion on where among the checks to do this, so I'll move it to the start.